ci/schema/duckdb.sql

```diff
@@ -51,3 +51,11 @@ CREATE OR REPLACE TABLE map (idx BIGINT, kv MAP(STRING, BIGINT));
 INSERT INTO map VALUES
     (1, MAP(['a', 'b', 'c'], [1, 2, 3])),
     (2, MAP(['d', 'e', 'f'], [4, 5, 6]));
+
+INSTALL spatial;
+LOAD spatial;
```

Comment:
I added this because the sqlalchemy duckdb compatibility job was failing. Is that the right way to go about it?

Reply:
I don't think we want this here. We should be auto loading the extension. If removing this line makes a test fail then we should figure out where we're missing the extension check.

Reply:
I remove the lines, and we get the SQLAlchemy 2 DuckDB job failing.
 
I think what's happening is that the data on the `duckdb.sql` get loaded here https://github.com/ibis-project/ibis/blob/4780fcb90f28af81dbd14c52b6228e8144412c6f/ibis/backends/duckdb/tests/conftest.py#L49-L72

which is relying on the `super().ddl_script` to create the table here https://github.com/ibis-project/ibis/blob/4780fcb90f28af81dbd14c52b6228e8144412c6f/ibis/backends/tests/base.py#L72-L80

and when it gets to the geospatial table it doesn't have the extension to do so. But it's not clear to me, where we should pass this info if it's not on the sql.

Reply:
Hmm, but the super call should come after we've loaded so spatial in the duckdb conftest.

Reply:
I'm on mobile and the test output won't load on my phone but I'll take a look when I'm back at a keyboard

Reply:
I'm guessing the nix failures are due to sandboxing and those can be skipped / xfailed the same way we did for the other tests

Reply:
But aren't we just loading the extension for the tables in TEST_TABLE_GEO? 

https://github.com/ibis-project/ibis/blob/4780fcb90f28af81dbd14c52b6228e8144412c6f/ibis/backends/duckdb/tests/conftest.py#L62-L71

Reply:
They all get run in the same session so it shouldn't matter. But try adding spatial to the extension list in `preload` and see if that sorts it out

Reply:
>I'm guessing the nix failures are due to sandboxing and those can be skipped / xfailed the same way we did for the other tests

Maybe I'm missing something but the other tests on `test_geospatial.py` we didn't have to do the skip due to sandbox. But I'm seeing a lot of failures now on the nix jobs, because it's trying to create the geo table and it's missing the extension. Let's look at this later in today when you are hands on keyboard.

Reply:
Ok.  I'm not entirely sure how this was working before (with the existing `st_read` calls), but now we don't load any `geospatial` stuff if were in a nix build.